src/python/turicreate/toolkits/one_shot_object_detector/util/_augmentation.py

```diff
@@ -39,7 +39,12 @@ def preview_synthetic_training_data(data,
     out : SFrame
         An SFrame of sythetically generated annotated training data.
     """
+
     dataset_to_augment, image_column_name, target_column_name = check_one_shot_input(data, target)
+    original_dataset_size = len(dataset_to_augment)
+    dataset_to_augment = dataset_to_augment.dropna(columns=image_column_name)
```

Comment:
If this is a common operation. Might be worth turning into a toolkit utility function.

Reply:
FYI, just from code inspection, I see that supervised learning toolkits either error out to use NAN for missing values (silently). They have user-controllable options for e.g. `predict` and `evaluate` but none of these drop any rows.

Reply:
We don't have consistency for this.

For example, the logistic classifier errors out with the error message "Missing value (None) encountered in column 'column_name'. Use the SFrame's dropna function to drop rows with 'None' values in them."
Random forest silently uses NAN values for missing values.
Image Classifier fails with the error message "ToolkitError: Cannot resize non-image type" because the image extractor can probably not handle None values. 
Drawing classifier uses exactly this since I put this in before WWDC.

Reply:
To close the loop from offline discussion: let's be consistent with the oldest, most testing toolkits and error out.

Given the shared part of this code is really just the error message, I wouldn't bother factoring it out, especially in Python code that is going away eventually